work_queue/src/work_queue.c

```diff
@@ -3094,13 +3099,16 @@ Used for computing queue capacity below.
 static void add_task_report(struct work_queue *q, struct work_queue_task *t)
 {
 	struct work_queue_task_report *tr;
+	struct work_queue_stats s;
+	work_queue_get_stats(q, &s);
 
 	// Create a new report object and add it to the list.
 	tr = calloc(1, sizeof(struct work_queue_task_report));
 
 	tr->transfer_time = (t->time_when_commit_end - t->time_when_commit_start) + (t->time_when_done - t->time_when_retrieval);
 	tr->exec_time     = t->time_workers_execute_last;
-
+	tr->master_time   = (((t->time_when_done - t->time_when_commit_start) - tr->transfer_time) - tr->exec_time);
```

Comment:
I like this definition much better. Please remove the dangling commented code below.